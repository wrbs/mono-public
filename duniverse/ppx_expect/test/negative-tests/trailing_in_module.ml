open! Core

module M (X : Sexpable) = struct
  module N (Y : sig
    val x : X.t
  end) =
  struct
    let%expect_test "trailing output" =
      let sexp = X.sexp_of_t Y.x in
      print_s sexp
    ;;

    let%expect_test "error" =
      if String.length (string_of_sexp (X.sexp_of_t Y.x)) > 1
      then raise_s (Base.Sexp.message "sexp is too long" [ "input: ", X.sexp_of_t Y.x ])
    ;;
  end
end

module String_tests = M (String)

module Run_on_abc = String_tests.N (struct
  let x = "a\nb\nc"
end)
