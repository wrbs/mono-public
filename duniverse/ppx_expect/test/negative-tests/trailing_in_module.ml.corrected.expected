open! Core

module M (X : Sexpable) = struct
  module N (Y : sig
      val x : X.t
    end) =
  struct
    let%expect_test "trailing output" =
      let sexp = X.sexp_of_t Y.x in
      print_s sexp;
      [%expect
        {|
         "a\
        \nb\
        \nc"
        |}]
    ;;

    let%expect_test "error" =
      if String.length (string_of_sexp (X.sexp_of_t Y.x)) > 1
      then raise_s (Base.Sexp.message "sexp is too long" [ "input: ", X.sexp_of_t Y.x ])
    [@@expect.uncaught_exn
      {|
      ("sexp is too long" ("input: "  "a\
                                     \nb\
                                     \nc"))
      |}]
    ;;
  end
end

module String_tests = M (String)

module Run_on_abc = String_tests.N (struct
    let x = "a\nb\nc"
  end)
